Source/CompSpinalEngineTrail.cs

```diff
@@ -0,0 +1,179 @@
+            {
+                cachedError = value;
+            }
+        }
+        public int PowerUse
+        {
+            get
+            {
+                if (parent.TryGetComp<CompSpinalEngineTrail>() == null)
+                {
+                    Log.Message("SOS2 spinal engines: somehow hit powerUse from a non-compspinalenginetrail?");
+                    return (int)(parent.TryGetComp<CompPowerTrader>().Props.PowerConsumption); // pretty? Sure this cannot happen but in case
+                }
+                if (fullyFormed)
+                { 
+                    return 0 - cachedPowerUse; 
+                }
+                else
+                {
+                    return (int)(parent.TryGetComp<CompPowerTrader>().Props.PowerConsumption);
+                }
+            }
+        }
+        public override int FuelUse
+        {
+            get
+            {
+                if (parent.TryGetComp<CompSpinalEngineTrail>() == null) //should never? happen but reflex
+                {
+                    Log.Message("SOS2 spinal engines: somehow hit FuelUse from a non-compspinalenginetrail?");
+                    return base.FuelUse;
+                }
+                if (fullyFormed)
+                {
+                    //Log.Message($"sos2 spinal engines thrust called about a spinal engine, returning cached fuelUse {cachedFuelUse}");
+                    return cachedFuelUse;
+                }
+                else
+                {
+                    //Log.Message($"sos2 spinal engines fuel called about a spinal engine, not fully formed");
+                    calc();
+                    return cachedFuelUse;
+                }
+            }
+        }
+        public int SupportWeight
+        {
+            get
+            {
+                if (parent.TryGetComp<CompSpinalEngineTrail>() == null) //should never? happen but reflex
+                {
+                    Log.Message("SOS2 spinal engines: somehow hit supportWeight from a non-compspinalenginetrail?!!!!??");
+                    return 0;// is this a log.error? Something went squirelly 
+                }
+                if (fullyFormed)
+                {
+                    return cachedMass;
+                }
+                else
+                {
+                    calc();
+                    return cachedMass;
+                }
+            }
+        }
+
+        public void Reset()
+        {
+            //Log.Message($"SOS2spinal engines: resetting: thrust {cachedThrust} mass {cachedMass} fueluse {cachedFuelUse} fuelallowed {cachedFuelAllowed} power {cachedPowerUse}");
+            cachedThrust = 0;
+            cachedMass = 0;
+            cachedFuelUse = 0;
+            cachedFuelAllowed = 0;
+            cachedPowerUse = 0;
+            cachedError = "Spinal engine isn't fully formed! Check for engine accelerators, and all three components of the fuel infrastructure!";
+        }
+        private void calc()
```

Comment:
Huh VS argued with me that I should rename the other methods, and it does show this one as well but I missed it. I was under the impression that a common style however was that *public* things would start with an uppercase, but I never checked this since it's a private method? Is there a common style folks use there?

Reply:
[Microsoft's C# style guide](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names)
says to "Use PascalCase for class names and method names." Since it doesn't specify scoping I take it to mean all class/method names.  That's been my experience working with other folks too.

All that said, being consistent in your style is the most important part.  And if you want to make private methods camelCase, go for it :)